apiserver/vshn/postgres/multiwatch.go

```diff
@@ -0,0 +1,44 @@
+package postgres
+
+import "k8s.io/apimachinery/pkg/watch"
+
+// MultiWatch is wrapper of multiple source watches which implements the same methods as a normal watch.Watch
+var _ watch.Interface = &MultiWatcher{}
+
+type MultiWatcher struct {
+	watchers  []watch.Interface
+	eventChan chan watch.Event
+}
+
+// NewEmptyMultiWatch creates an empty watch
+func NewEmptyMultiWatch() *MultiWatcher {
+	return &MultiWatcher{
+		eventChan: make(chan watch.Event),
+	}
+}
+
+// AddWatcher adds a watch to this MultiWatcher
+func (m *MultiWatcher) AddWatcher(w watch.Interface) {
+	m.watchers = append(m.watchers, w)
+}
+
+// Stop stops all watches of this MultiWatch
+func (m *MultiWatcher) Stop() {
+	for _, watcher := range m.watchers {
+		watcher.Stop()
+	}
+	close(m.eventChan)
+}
```

Comment:
This is nitpicky but I think there is a small chance that this will cause a panic.

1. A proxy goroutine pulls an event form the upstream channel
2. `Stop` is called. All channels are closed
3. The proxy goroutine continues, tries to write to a closed channel, and panics

Not sure if this actually ever happens.

Reply:
do you expect some locks here?

Reply:
A WaitGroup might make sense so that we wait for all goroutines to finish before closing the channel?

Reply:
Just added it, can you check again?